internal/configs/parsing_helpers.go

```diff
@@ -162,31 +162,69 @@ func validateHashLBMethod(method string) (string, error) {
 	return "", fmt.Errorf("Invalid load balancing method: %q", method)
 }
 
-// http://nginx.org/en/docs/syntax.html
-var validTimeSuffixes = []string{
-	"ms",
-	"s",
-	"m",
-	"h",
-	"d",
-	"w",
-	"M",
-	"y",
+// ParseBool ensures that the string value is a valid bool
+func ParseBool(s string) (bool, error) {
+	return strconv.ParseBool(s)
 }
 
-var durationEscaped = strings.Join(validTimeSuffixes, "|")
-var validNginxTime = regexp.MustCompile(`^([0-9]+([` + durationEscaped + `]?){0,1} *)+$`)
+// ParseInt ensures that the string value is a valid int
+func ParseInt(s string) (int, error) {
+	return strconv.Atoi(s)
+}
+
+// ParseInt64 ensures that the string value is a valid int64
+func ParseInt64(s string) (int64, error) {
+	return strconv.ParseInt(s, 10, 64)
+}
+
+// ParseUint64 ensures that the string value is a valid uint64
+func ParseUint64(s string) (uint64, error) {
+	return strconv.ParseUint(s, 10, 64)
+}
+
+// timeRegexp http://nginx.org/en/docs/syntax.html
+var timeRegexp = regexp.MustCompile(`^([0-9]+([ms|s|m|h|d|w|M|y]?){0,1} *)+$`)
```

Comment:
Building the regexp from the slice of time units didn't seem to be worth the effort.